src/main/java/com/BlogSite/TestBlogProject/services/PostServiceImpl.java

```diff
@@ -39,15 +38,15 @@ public List<Post> getAllPosts() {
     public Result<Post> addPost(PostDto postDto) {
         Result<Post> result = new Result<>();
         Result<User> userResult = userService.getUserByUsername(postDto.getUsername());
-        if (userResult.getError() == ErrorCode.USER_NOT_FOUND) {
+        if (userResult.getError() == ErrorCode.USER_NOT_FOUND)
             result.setError(userResult.getError());
-            return result;
+        else {
+            Post post = new Post();
```

Comment:
highly recommended to use mappers. Pay your attention to mapstruct or selma mappers.

https://www.baeldung.com/java-performance-mapping-frameworks